en/dev_setup/dev_env_windows_wsl.md

```diff
@@ -0,0 +1,238 @@
+1. Execute the command `wsl` to access the WSL shell.
+1. WSL will prompt you for a user name and password for the Ubuntu installation.
+   **Note:** _Remember the credentials for later on!_
+
+:::note
+If you have any problems with your setup, check the current microsoft WSL installation documentation.
+:::
+
+### Install PX4 Toolchain
+
+To install the development toolchain for Gazebo simulation, JMAVSim simulation and Pixhawk/NuttX harware:
+
+1. Execute the command `cd ~` to switch to the home folder of WSL for the next steps.
+   :::warning
+   This is important!
+   If you work from a location outside of the WSL file system you'll run into issues such as very slow execution and access right/permission errors.
+   :::
+
+1. [Download PX4 Source Code](../dev_setup/building_px4.md):
+   ```bash
+   git clone https://github.com/PX4/PX4-Autopilot.git --recursive
+   ```
+1. Run the **ubuntu.sh** installer script and acknowledge any prompts as the script progresses:
+   ```bash
+   bash ./PX4-Autopilot/Tools/setup/ubuntu.sh
+   ```
+   
+   :::note
+   This installs tools to build PX4 for Pixhawk, Gazebo and JMAVSim targets.
+   - You can use the `--no-nuttx` and `--no-sim-tools` options to omit the NuttX and/or simulation tools.
+   - Other Linux build targets are untested (you can try these by entering the appropriate commands in [Ubuntu Development Environment](../dev_setup/dev_env_linux_ubuntu.md) into the WSL shell).
+     Note that `git` is already installed in WSL.
+   
+1. Restart the "WSL computer" after the script completes:
+   - Exit the shell and shut down WSL:
+     ```
+     exit
+     wsl --shutdown
+     ```
+     (or do `exit`, close the prompt, and reopen a new one).
+   - Start WSL again and access the shell:
+     ```
+     wsl
+     ```
+1. Switch to the PX4 repository in the WSL home folder:
+   ```
+   cd ~/PX4-Autopilot
+   ```
+1. Build the PX4 SITL target and test your environment:
+   ```
+   make px4_sitl
+   ```
+
+
+### Opening a WSL shell
+
+To open the WSL shell (once WSL has been installed):
+
+1. Press the Windows **Start** key.
+1. Type "cmd" and press Enter to open a command prompt.
+1. Execute the command `wsl` to access the WSL shell.
+
+
+## Visual Studio Code Integration
+
+1. Download and install Visual Studio Code (VS Code) on Windows from <https://code.visualstudio.com/>
+2. Open _Visual Studio Code_.
+3. Install the extension called [Remote - WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) (marketplace) 
+4. [Open a WSL shell](#opening-a-wsl-shell)
+5. In the WSL shell, switch to the PX4 folder:
+   ```
+   cd ~/PX4-Autopilot
+   ```
+6. In the WSL shell, start VS Code:
+   ```
+   code .
+   ```
+   This will open the IDE fully integrated with the WSL shell.
+   Make sure you always open the PX4 repository in the Remote WSL mode.
+   It will be available in your recently opened locations whenever you start VS Code.
```

Comment:
I get items 5, 6 now - after running VS Code using `code .` it is associatd with the WSL shell. Further, you're saying if you open recent the option will be present to start this again. Is that right?

To be very clear, if I turned by computer on and off and came back to VS code WSL would have closed down. Are you saying that selecting recent would reopen WSL with the correct integration? What about things like the IP address - even if I reopen WSL2 I no longer have knowledge of what that is. Is it that if I open WSL in a prompt again it will open the same WSL as VS Code does?

Reply:
> But what if I turn my computer on/off and came back to VS code. WSL would have closed down.

Yes, it shuts down the WSL kernel when you turn off your computer (or run `wsl --shutdown` in a Windows shell).

> Are you saying that selecting the option would reopen WSL with the correct integration?

Yes, as soon as you access one of the repositories you're working on through WSL e.g.
![image](https://user-images.githubusercontent.com/4668506/158422116-d10b4158-a981-48dd-9b7a-eded9de6382e.png)
it will automatically start WSL and open the repository with the same setup you had open when you last left.

> What about things like the IP address - even if I reopen WSL2 I no longer have knowledge of what that is.

The IP address of WSL changes with every restart. VS code does all the magic in the background but for the ground station that is not running within WSL you have to manually check the IP and add it to the "comm link" of QGC.

> Is it that if I open WSL in a prompt again it will open the same WSL as VS Code does?

Yes, if you run the `wsl` command on a Windows shell it will enter a shell on the same WSL shell as VSCode is using.

Note: You can in principle have multiple WSLs and even different distributions but I never bothered. Then you have to always specify what opens where.
Note2: You can also open the repository which is in WSL directly in VSCode (just the folder) but then you don't have all the advantages of the linux like experience within the IDE. That's why it matters how you open it all the way at the begining and afterwards you can just "open recent" and reuse the setup.

Reply:
Excellent. Clear now.